backend/src/controllers/user.controller.ts

```diff
@@ -0,0 +1,86 @@
+import { Request, Response } from "express";
+import {
+  create,
+  deleteUser,
+  findAll,
+  findById,
+  update,
+} from "../services/user.service";
+
+export const getAllUsers = async (req: Request, res: Response) => {
+  try {
+    const users = await findAll();
+    return res.send(users).status(200);
+  } catch (error) {
+    console.error(error);
+    return res.status(500).send("Internal Server Error");
+  }
+};
+
+export const getUserById = async (req: Request, res: Response) => {
+  const userId: number = parseInt(req.params.id);
+
+  try {
+    const user = await findById(userId);
+
+    if (!user) {
+      return res.status(404).send("User not found");
+    }
+
+    return res.send(user).status(200);
+  } catch (error) {
+    console.error(error);
```

Comment:
centralize this catch statement, since u have console.error almost in each catch block. Better to call dedicated function which will log stuff and do the return.